velox/vector/ComplexVector.cpp

```diff
@@ -96,7 +96,7 @@ std::optional<int32_t> RowVector::compare(
   auto compareSize = std::min(children_.size(), otherRow->children_.size());
   for (int32_t i = 0; i < compareSize; ++i) {
     BaseVector* child = children_[i].get();
-    BaseVector* otherChild = otherRow->childAt(i)->loadedVector();
```

Comment:
wouldn't this mean to is this is a lazy vector, it would not be loaded first? Cc: @bikramSingh91 who understands this codebase much better than I do